tests_new/integration_tests/modules/redshift_datasets/test_redshift_connection.py

```diff
@@ -0,0 +1,362 @@
+    ).contains(
+        'UnauthorizedOperation',
+        'MANAGE_REDSHIFT_CONNECTION_PERMISSIONS',
+        session_connection_serverless_admin.connectionUri,
+    )
+
+
+def test_delete_connection_group_permissions_invalid_connection_type(
+    client1, group3, session_connection_serverless_data_user
+):
+    assert_that(delete_redshift_connection_group_permissions).raises(GqlError).when_called_with(
+        client=client1,
+        connection_uri=session_connection_serverless_data_user.connectionUri,
+        group_uri=group3,
+    ).contains('InvalidInput', 'ConnectionType', session_connection_serverless_data_user.connectionType)
+
+
+def test_delete_connection_group_permissions_invalid_group(client1, group1, session_connection_serverless_admin):
+    assert_that(delete_redshift_connection_group_permissions).raises(GqlError).when_called_with(
+        client=client1,
+        connection_uri=session_connection_serverless_admin.connectionUri,
+        group_uri=group1,
+    ).contains('InvalidInput', 'Team', group1, 'EXCEPT the connection owners')
+
+
+def test_list_redshift_environment_connections(
+    client1, group1, session_env1, session_connection_serverless_admin, session_connection_serverless_data_user
+):
+    response = list_environment_redshift_connections(
+        client=client1,
+        environment_uri=session_env1.environmentUri,
+        group_uri=group1,
+    )
+    assert_that(response.count).is_equal_to(2)
+    assert_that(response.nodes).extracting('connectionUri').contains(
+        session_connection_serverless_admin.connectionUri, session_connection_serverless_data_user.connectionUri
+    )
+    response = list_environment_redshift_connections(
+        client=client1,
+        environment_uri=session_env1.environmentUri,
+        group_uri=group1,
+        connection_type='DATA_USER',
+    )
+    assert_that(response.count).is_equal_to(1)
+    assert_that(response.nodes).extracting('connectionUri').contains(
+        session_connection_serverless_data_user.connectionUri
+    )
+    assert_that(response.nodes).extracting('connectionUri').does_not_contain(
+        session_connection_serverless_admin.connectionUri
+    )
+
+
+def test_list_redshift_environment_connections_unauthorized(client2, group1, session_env1):
+    assert_that(list_environment_redshift_connections).raises(GqlError).when_called_with(
+        client=client2,
+        environment_uri=session_env1.environmentUri,
+        group_uri=group1,
+    ).contains(
+        'UnauthorizedOperation',
+        'LIST_ENVIRONMENT_REDSHIFT_CONNECTIONS',
+        session_env1.environmentUri,
+    )
+
+
+def test_list_redshift_connection_schemas(client1, session_connection_serverless_admin):
+    response = list_redshift_connection_schemas(
+        client=client1, connection_uri=session_connection_serverless_admin.connectionUri
+    )
+    assert_that(len(response)).is_greater_than_or_equal_to(1)
+    assert_that(response).contains('public')
```

Comment:
nit: make `public` a constant defined at top of file

Reply:
Done, added to glob_conftest